src/commands/lobby.ts

```diff
@@ -1,50 +1,75 @@
-import { ActionRowBuilder, ButtonBuilder, ButtonStyle } from "discord.js";
+import { ComponentType } from "discord.js";
 import GameManager from "../util/GameManager/GameManager.js";
 import { fetchMember } from "../util/MemberUtil/MemberFetch.js";
+import LobbyEmbed from "../util/Embeds/LobbyEmbed.js";
 
 const __lobby: InteractionHandlerPayloads.GuildChatInputCommand = {
     name: "start",
     description: "Start a Black Jack Game",
 
     async execute(interaction) {
         const gameManager = new GameManager(interaction);
+        const lobbyEmbed = new LobbyEmbed(gameManager.players);
 
-        //test add player
-        gameManager.players.addPlayer(
-            await fetchMember(interaction.guildId, interaction.member.id),
-            interaction.member
-        );
-
-        // get the lobby emebed
-        const embed = gameManager.LobbyEmbed();
-
-        // creates the join, leave and start now buttons
-        const actionRow = new ActionRowBuilder<ButtonBuilder>().addComponents(
-            new ButtonBuilder()
-                .setCustomId("join-game")
-                .setLabel("Join")
-                .setStyle(ButtonStyle.Success),
-
-            new ButtonBuilder()
-                .setCustomId("leave-game")
-                .setLabel("Leave")
-                .setStyle(ButtonStyle.Danger),
-
-            new ButtonBuilder()
-                .setCustomId("start-game")
-                .setLabel("Start Now")
-                .setStyle(ButtonStyle.Primary)
-        );
-
-        await interaction.reply({
+        console.log("A lobby has been created");
+
+        // create and display the lobby emebed
+        const { embeds: embed, components: actionRow } =
+            lobbyEmbed.CreateEmbed();
+
+        const message = await interaction.reply({
             embeds: [embed],
             components: [actionRow],
         });
 
+        const collector = message.createMessageComponentCollector({
+            filter: (i) => i.user.id === interaction.user.id,
+            componentType: ComponentType.Button,
+            time: 120_000,
+        });
+
+        // collector that responds to the buttons
+        collector.on("collect", async (i) => {
+            if (i.customId === "join-game") {
+                // add player if not already in the lobby
+                if (!gameManager.players.getPlayer(interaction.member.id)) {
```

Comment:
Turn the if statement into a guard clause: Inverse the logic to player already in the lobby and "return" out of the callback function. This avoids the additional nest. Line 44 and 45 should only update if there's a change, so make sure they're covered by the guard clause as well.